src/main/java/org/sonarlint/intellij/analysis/SonarLintTask.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.analysis;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.sonarlint.intellij.editor.AccumulatorIssueListener;
+import org.sonarlint.intellij.issue.IssueProcessor;
+import org.sonarlint.intellij.ui.SonarLintConsole;
+
+public class SonarLintTask extends Task.Backgroundable {
+  private static final Logger LOGGER = Logger.getInstance(SonarlintAnalyzer.class);
+  private final IssueProcessor processor;
+  private final SonarlintAnalyzer.SonarLintJob job;
+  private final boolean startInBackground;
+
+  private SonarLintTask(IssueProcessor processor, SonarlintAnalyzer.SonarLintJob job, boolean background) {
+    super(job.module().getProject(), "SonarLint Analysis", true);
+    this.processor = processor;
+    this.job = job;
+    this.startInBackground = background;
+  }
+
+  public static SonarLintTask createBackground(IssueProcessor processor, SonarlintAnalyzer.SonarLintJob job) {
+    return new SonarLintTask(processor, job, true);
+  }
+
+  public static SonarLintTask createForeground(IssueProcessor processor, SonarlintAnalyzer.SonarLintJob job) {
+    return new SonarLintTask(processor, job, false);
+  }
+
+  @Override
+  public boolean shouldStartInBackground() {
+    return startInBackground;
+  }
+
+  private void stopRun() {
+    SonarLintStatus.get(job.module().getProject()).stopRun();
+  }
+
+  private static String getFileName(VirtualFile file) {
+    return file.getName();
+  }
+
+  @Override
+  public void run(ProgressIndicator indicator) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "run" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)